selenium/che-selenium-test/src/test/java/org/eclipse/che/selenium/hotupdate/RolloutUpdateStrategyWithEditorTest.java

```diff
@@ -0,0 +1,158 @@
+  private static final String COMMAND_FOR_GETTING_CURRENT_DEPLOYMENT_CHE =
+      "get dc | grep che | awk '{print $2}'";
+
+  private static final String NAME_OF_CHECKED_CLASS = "AppController";
+  private static final String TEXT_FOR_TYPING = "test\n";
+  private static final String EXPECTED_DEFAULT_TEXT =
+      "/*\n"
+          + " * Copyright (c) 2012-2018 Red Hat, Inc.\n"
+          + " * All rights reserved. This program and the accompanying materials\n"
+          + " * are made available under the terms of the Eclipse Public License v1.0\n"
+          + " * which accompanies this distribution, and is available at\n"
+          + " * http://www.eclipse.org/legal/epl-v10.html\n"
+          + " *\n"
+          + " * Contributors:\n"
+          + " *   Red Hat, Inc. - initial API and implementation\n"
+          + " */\n"
+          + "package org.eclipse.qa.examples;\n"
+          + "\n"
+          + "import java.util.Random;\n"
+          + "\n";
+
+  private static final String EXPECTED_CHANGED_TEXT =
+      "test\n"
+          + "/*\n"
+          + " * Copyright (c) 2012-2018 Red Hat, Inc.\n"
+          + " * All rights reserved. This program and the accompanying materials\n"
+          + " * are made available under the terms of the Eclipse Public License v1.0\n"
+          + " * which accompanies this distribution, and is available at\n"
+          + " * http://www.eclipse.org/legal/epl-v10.html\n"
+          + " *\n"
+          + " * Contributors:\n"
+          + " *   Red Hat, Inc. - initial API and implementation\n"
+          + " */\n"
+          + "package org.eclipse.qa.examples;\n"
+          + "\n"
+          + "import java.util.Random;\n"
+          + "\n";
+
+  @Inject private Wizard projectWizard;
+  @Inject private Menu menu;
+  @Inject private ProjectExplorer projectExplorer;
+  @Inject private Consoles console;
+  @Inject private CodenvyEditor editor;
+  @Inject private Ide ide;
+  @Inject private TestWorkspace workspace;
+  @Inject private TestProjectServiceClient testProjectServiceClient;
+  @Inject private WebDriverWaitFactory webDriverWaitFactory;
+  @Inject private CheTestSystemClient cheTestSystemClient;
+  @Inject private OpenShiftCliCommandExecutor openShiftCliCommandExecutor;
+
+  @BeforeMethod
+  public void prepare() throws Exception {
+    Path pathToProject =
+        Paths.get(getClass().getResource("/projects/default-spring-project").toURI());
+
+    testProjectServiceClient.importProject(
+        workspace.getId(), pathToProject, PROJECT_NAME, MAVEN_SPRING);
+  }
+
+  @Test
+  public void shouldUpdateMasterByRolloutStrategyWithAccessibleEditorInProcess() throws Exception {
+    int currentRevision = getRevision();
+
+    // check editor availability
+
+    ide.open(workspace);
+    projectExplorer.waitProjectExplorer();
+    projectExplorer.waitItem(PROJECT_NAME);
+
+    projectExplorer.quickExpandWithJavaScript();
+
+    projectExplorer.openItemByVisibleNameInExplorer(NAME_OF_CHECKED_CLASS + ".java");
```

Comment:
We can avoid duplication if we will have dedicated constant to describe `NAME_OF_CHECKED_CLASS_FILE = NAME_OF_CHECKED_CLASS + ".java"`